File: sdk-info/src/main/java/com/itsaky/apiinfo/models/Info.java
package com.itsaky.apiinfo.models;

/**
 * Base class for information about a class, method, field
 */
public abstract class Info {
    
    /**
     * Name of this class, method or field
     */
    public String name = "";
    
    /**
     * API level in which this info was deprecated
     */
    public int deprecated = -1;
    
    /**
     * API level in which this was added
     */
    public int since = -1;
    
    /**
     * API level in which this was removed
     */
    public int removed = -1;
}

File: build-tools/xml-utils/src/main/java/com/itsaky/androidide/xml/versions/Info.kt
/*
 *  This file is part of AndroidIDE.
 *
 *  AndroidIDE is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  AndroidIDE is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *   along with AndroidIDE.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.itsaky.androidide.xml.versions

/**
 * Base class for info about class, fields and methods.
 *
 * @author Akash Yadav
 */
interface Info {

  /** Name of this element. */
  val name: String

  /** The introducing API version. */
  val since: Int

  /** The removing API version. */
  val removed: Int

  /** The deprecating API version. */
  val deprecated: Int
}
